### ISSUE TITLE
Allegro doesn't emit ALLEGRO_EVENT_KEY_CHAR on Ubuntu 14.04+

### ISSUE THREAD

**Issue by oitofelix** [liballeg/allegro5#720]
When testing [MININIM](http://oitofelix.github.io/mininim/) in recent versions of Ubuntu, I was surprised by the fact that it can't detect most of keyboard input.  Investigating this issue further I discovered that replacing

```
if (!filtered && mycode < ALLEGRO_KEY_MODIFIERS) {
```

by

```
if (/* !filtered && */ mycode < ALLEGRO_KEY_MODIFIERS) {
```

in [src/x/xkeyboard.c:1050](https://github.com/liballeg/allegro5/blob/53a775e1c6aec1b0d67563a21682d9cf36c139fe/src/x/xkeyboard.c#L1050) made it work as expected.  Thus this behavior is related to input methods (more specifically [XFilterEvent](http://www.x.org/pub/current/doc/man/man3/XFilterEvent.3.xhtml) returning true).

I'm not sure on how to properly solve this problem, or even whether it's Allegro's fault at all.  I can say however, that many Allegro games will not work properly out of the box in Ubuntu (the most popular GNU/Linux distribution).  Anyway, I should work around this in my game using ALLEGRO_EVENT_KEY_DOWN, which still works fine.

**Comment by oitofelix** [comment]
I guess ALLEGRO_EVENT_KEY_DOWN doesn't work that fine after all.  See issue #721.

**Comment by allefant** [comment]
I believe this has been fixed by #5880fb2
